src/services/mongoose/wallets.ts

```diff
@@ -7,6 +7,27 @@ import { User } from "@services/mongoose/schema"
 import { caseInsensitiveRegex } from "./users"
 
 export const WalletsRepository = (): IWalletsRepository => {
+  const persistNewOnChainAddress = async (
```

Comment:
Can we stick to the:
```
wallet = repo.findById(id)
wallet.onChainAddressIdentifieds.push(wallet)
repo.update(wallet)
```
Pattern?

ie Repo is only used for retrieving and persisting _not_ updating specific fields? There are exceptions, like bulk update to avoid n+1 problem, but that doesn't apply here.

Reply:
if I do this then I will need to use a lock on user  (to avoid addresses lost), also we will loose previous ids and will affect re-indexing

so.. maybe another repo is better just for onchain addresses, what do you think? (we will migrate to this later)

Reply:
I like the idea to have a repo just for `onchainAddresses`!